### PR TITLE
Developer mode

### PR DESCRIPTION
**Pull request by mdtanrikulu**
To activate/deactivate developer mode; 

Windows, Linux:
<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd>

Mac: 
<kbd>cmd⌘</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd>

To activate/deactivate debug mode; 

Windows, Linux:
<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>L</kbd>

Mac: 
<kbd>cmd⌘</kbd> + <kbd>Shift</kbd> + <kbd>L</kbd>

### CODE REVIEW

File: electron/config_storage.js [hunk 1]
```diff
@@ -1,40 +1,48 @@
-var electron = window.require('electron');
-const Config = electron.remote.require('electron-config')
+var electron = require('electron');
```
Comment (maaktweluit) [inline electron/config_storage.js]: any reason not to use `const` here?

File: electron/config_storage.js [hunk 2]
```diff
@@ -1,40 +1,48 @@
...
  * [setConfig storing key value config persistent]
- * @param {[Any]} 		key  
- * @param {[Any]} 		value 
+ * @param {[Any]}       key  
+ * @param {[Any]}       value 
  *
  * @see https://github.com/sindresorhus/electron-config 
  */
-export function setConfig(key, value) {
+let setConfig = function(key, value) {
```
Comment (maaktweluit) [inline electron/config_storage.js]: could keep it simple with just
`function setConfig(...`
same for getter

File: electron/config_storage.js [hunk 3]
```diff
@@ -1,40 +1,48 @@
...
     return config.get(key)
 }
 
+const configStorage = {
+    setConfig,
+    getConfig,
+    dictConfig
+}
 
+module.exports = global.configStorage = configStorage
```
Comment (maaktweluit) [inline electron/config_storage.js]: no newline at end of file is bad for git diffs

File: electron/debug_handler.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    })
+}
+
+const logger = Object.freeze(Object.entries(level)
+    .reduce((total, [_, value]) => {
+        total[value] = callback.bind(null, value)
+        return total
+    }, {}))
+
+module.exports = global.log = logger
```
Comment (maaktweluit) [inline electron/debug_handler.js:55]: white line at eof

File: electron/golem_handler.js [hunk 5]
```diff
@@ -2,6 +2,7 @@ const electron = require('electron');
 const fs = require('fs');
 const os = require('os');
 const path = require('path');
+const log = require('./debug_handler.js')
```
Comment (maaktweluit) [inline electron/golem_handler.js:5]: we can `require().getLogger('FILE_IDENTIFIER')` to not repeat the `'MAIN_PROCESS > GOLEM_HANDLER'` later

`getLogger` is pseudo code, not sure if it exists ( yet ;) )

File: index.js [hunk 6]
```diff
@@ -7,7 +7,9 @@ var mkdirp = require('mkdirp');
 
 //require('electron-debug')({showDevTools: true, enabled: true});
 
-const createTray = require('./electron/tray_handler.js')
+//const createTray = require('./electron/tray_handler.js')
```
Comment (maaktweluit) [inline index.js]: commented out to stay? we used to only allow commented out code if it has an explanation comment.

File: src/components/tasks/TaskDetail.js [hunk 7]
```diff
@@ -206,14 +221,14 @@ export class TaskDetail extends React.Component {
     }
 
     _convertPriceAsHR(price) {
-        console.log("price", price);
+        // console.log("price", price);
```
Comment (maaktweluit) [inline src/components/tasks/TaskDetail.js]: logs to stay? could become `log.debug` or `log.silly`

File: src/reducers/frame/single.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { dict } from './../../actions'
 
-const {SET_FRAMES_WITH_SUBTASKS, SET_SUBTASKS_BORDER, SET_PREVIEW_LIST, SET_SUBTASKS_LIST, SET_SUBTASKS_VISIBILITY, SET_FRAME_ID, NEXT_FRAME, PREVIOUS_FRAME} = dict
+const {SET_FRAMES_WITH_SUBTASKS, SET_SUBTASKS_BORDER, SET_PREVIEW_LIST, SET_SUBTASKS_LIST, SET_SUBTASKS_VISIBILITY, SET_FRAME_ID, NEXT_FRAME, PREVIOUS_FRAME, CLEAR_TASK_PLAIN} = dict
```
Comment (maaktweluit) [inline src/reducers/frame/single.js]: maybe break down long line?

File: src/sagas/frame/index.js [hunk 9]
```diff
@@ -161,6 +161,7 @@ export function fetchSubtaskList(session, payload) {
 }
 
 export function* subtaskList(session, payload) {
+    console.log("payload", payload);
```
Comment (maaktweluit) [inline src/sagas/frame/index.js]: `log.verbose` or more debuggy?

File: src/sagas/stats.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+/**
+ * [*performance generator]
+ * @param  {Object} session     [Websocket connection session]
+ * @yield   {Object}            [Action object]
+ */
+export function* statsFlow(session) {
+    yield fork(fireBase, session)
+}
```
Comment (maaktweluit) [inline src/sagas/stats.js]: eof

**Review by maaktweluit — CHANGES_REQUESTED** [review]
sorry for not focusing much on the app / logic / flows, i'm new to this application so can not see the bigger picture yet.

Most comments are small and good practice proposals I have learned from the past.

File: src/sagas/frame/index.js (hunk 9 above)
Comment (mdtanrikulu) [reply src/sagas/frame/index.js]: debug

File: electron/golem_handler.js (hunk 5 above)
Comment (mdtanrikulu) [reply electron/golem_handler.js:5]: nice idea! I'll implement it soon.

**Comment by badb** [comment]
@mdtanrikulu Can we merge this, pleaaaase?

**Comment by mdtanrikulu** [comment]
Sure, let me check after merging dev into developer mode, then I'll merge it.

**Pull request merged**